metadata/metadata.go

```diff
@@ -157,6 +158,10 @@ func ExtractMeta(data []byte, spaceFromCli string, titleFromH1 bool, parents []s
 		if titleFromH1 && meta.Title == "" {
 			meta.Title = ExtractDocumentLeadingH1(data)
 		}
+		if titleFromFilename && meta.Title == "" && filename != "" {
+			base := filepath.Base(filename)
+			meta.Title = strings.TrimSuffix(base, filepath.Ext(base))
```

Comment:
Should this use https://pkg.go.dev/golang.org/x/text/cases#Title and do you think it's useful to replace underscores and with spaces? so `my_new_post.md` becomes `My New Post`